server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailImportContract.scala

```diff
@@ -0,0 +1,924 @@
+
+    assertThatJson(response)
+      .whenIgnoringPaths("methodResponses[0][1].oldState", "methodResponses[0][1].newState", "methodResponses[1][1].state")
+      .inPath("methodResponses[0]")
+      .isEqualTo(
+      s"""[
+           |  "Email/import",
+           |  {
+           |    "accountId":"$ACCOUNT_ID",
+           |    "created":{
+           |      "C42":{
+           |        "id":"$messageId",
+           |        "blobId":"$messageId",
+           |        "threadId":"$messageId",
+           |        "size":85
+           |      }
+           |    }
+           |  },
+           |  "c1"
+           |]""".stripMargin)
+  }
+
+  @Test
+  def importShouldFailWhenNoMailboxes(): Unit = {
+    val receivedAt = ZonedDateTime.now().minusDays(1)
+
+    val uploadResponse: String = `given`
+      .basePath("")
+      .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+      .body("whatever")
+    .when
+      .post(s"/upload/$ACCOUNT_ID/")
+    .`then`
+      .statusCode(SC_CREATED)
+      .extract
+      .body
+      .asString
+
+    val blobId: String = Json.parse(uploadResponse).\("blobId").get.asInstanceOf[JsString].value
+
+    val receivedAtString = UTCDate(receivedAt).asUTC.format(UTC_DATE_FORMAT)
+    val response = `given`
+      .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+      .body(s"""{
+              |  "using": ["urn:ietf:params:jmap:core", "urn:ietf:params:jmap:mail"],
+              |  "methodCalls": [
+              |    ["Email/import",
+              |      {
+              |        "accountId": "$ACCOUNT_ID",
+              |        "emails": {
+              |           "C42": {
+              |             "blobId": "$blobId",
+              |             "mailboxIds": {},
+              |             "keywords": {
+              |               "toto": true
+              |             },
+              |             "receivedAt": "$receivedAtString"
+              |           }
+              |         }
+              |      },
+              |      "c1"]
+              |  ]
+              |}""".stripMargin)
+    .when
+      .post
+    .`then`
+      .statusCode(SC_OK)
+      .contentType(JSON)
+      .extract
+      .body
+      .asString
+
+    assertThatJson(response)
+      .whenIgnoringPaths("methodResponses[0][1].oldState", "methodResponses[0][1].newState", "methodResponses[1][1].state")
+      .inPath("methodResponses[0]")
+      .isEqualTo(
+      s"""[
+           |  "Email/import",
+           |  {
+           |    "accountId":"29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+           |    "notCreated":{
+           |      "C42":{
+           |        "type":"invalidArguments",
+           |        "description":"Email/import so far only supports a single mailboxId"
+           |      }
+           |    }
+           |  },
+           |  "c1"
+           |]""".stripMargin)
+  }
+
+  @Test
+  def importShouldFailWhenTooManyMailboxes(): Unit = {
```

Comment:
so that's a limitation regarding the current RFC right? I think we support it for email/set create, so why not for the import?

Reply:
See https://issues.apache.org/jira/browse/JAMES-3538

We do not support it yet for Email/set create. Hence I was reluctant to support here too to be honest.

I prefer handling this in another issue.